Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp

```diff
@@ -1325,126 +1371,122 @@ namespace AZ::Data
-                newAssetData->RegisterWithHandler(handler);
-            }
-        }
+                // For reloaded assets, we need to hold an internal reference to ensure the data
+                // isn't immediately destroyed. Since reloads are not a shipping feature, we'll
+                // hold this reference indefinitely, but we'll only hold the most recent one for
+                // a given asset Id.
 
-        if (newAssetData)
-        {
-            // For reloaded assets, we need to hold an internal reference to ensure the data
-            // isn't immediately destroyed. Since reloads are not a shipping feature, we'll
-            // hold this reference indefinitely, but we'll only hold the most recent one for
-            // a given asset Id.
+                newAssetData->m_status = AssetData::AssetStatus::Queued;
+                newAsset = Asset<AssetData>(newAssetData, assetReferenceLoadBehavior);
 
-            newAssetData->m_status = AssetData::AssetStatus::Queued;
-            Asset<AssetData> newAsset(newAssetData, assetReferenceLoadBehavior);
+                m_reloads[newAsset.GetId()] = newAsset;
 
-            m_reloads[newAsset.GetId()] = newAsset;
+                UpdateDebugStatus(newAsset);
+            }
+        }
 
-            UpdateDebugStatus(newAsset);
+        AZStd::scoped_lock lock(m_assetContainerMutex);
+        container = GetAssetContainer(newAsset, {}, true);
 
-            AZStd::shared_ptr<AssetDataStream> dataStream;
-            AssetStreamInfo loadInfo = GetModifiedLoadStreamInfoForAsset(newAsset, handler);
-            constexpr bool isReload = true;
-            if (loadInfo.IsValid())
-            {
-                // Create the AssetDataStream instance here so it can claim an asset reference inside the lock (for a total
-                // count of 2 before starting the load), otherwise the refcount will be 1, and the load could be canceled
-                // before it is started, which creates state consistency issues.
+        if (container)
+        {
+            m_ownedAssetContainers.insert({ container.get(), container });
 
-                dataStream = AZStd::make_shared<AssetDataStream>(handler->GetAssetBufferAllocator());
-                if (dataStream)
-                {
-                    // Currently there isn't a clear use case for needing to adjust priority for reloads so the default load priority is used
-                    constexpr bool signalLoaded = false;    // this is a reload, so don't signal dependent-asset loads
-                    QueueAsyncStreamLoad(newAsset, dataStream, loadInfo, isReload,
-                        handler, {}, signalLoaded);
-                }
-                else
+            // Only insert a new entry into m_ownedAssetContainerLookup if one doesn't already exist for this container.
+            // Because it's a multimap, it is possible to add duplicate entries by mistake.
```

Comment:
"it is possible to add duplicate entries by mistake." strange question... then why is it a multi map?

Reply:
I've been wondering the same thing for a while now, I went to update this and finally realized why.  Its not that you can have multiple AssetId entries (the key for the map), its that you can accidentally record multiple instances of AssetIdA -> ContainerA.  An Asset can have multiple containers for it at the same time due to different load settings.  This is just trying to prevent duplicates of the value, not the key.

I've updated the comment to be more clear about this.